src/webots/gui/WbMainWindow.cpp

```diff
@@ -2475,8 +2475,10 @@ void WbMainWindow::prepareNodeRegeneration(WbNode *node) {
       QStringList perspective = factory->windowPerspective(device);
       if (perspective.isEmpty())
         perspective = device->perspective();
-      const WbRobot *robot = dynamic_cast<const WbRobot *>(WbNodeUtilities::findTopNode(node));
-      mTemporaryProtoPerspectives.insert(robot->name() + "\n" + device->name(), perspective);
+      const WbRobot *robot = WbNodeUtilities::findRobotAncestor(device);
+      assert(robot);
+      if (robot)
```

Comment:
Is it useful to check the robot variable here, as we already check it with the assertion above?

Reply:
Yes. The assertion is only executed in debug mode.
If we don't want that Webots crashes when running in release mode we need the check.